.github/workflows/reusable-workers-service-deploy.yml

```diff
@@ -0,0 +1,103 @@
+name: Deploy Workers Job
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+on:
+  workflow_call:
+    inputs:
+      environment:
+        required: true
+        type: string
+      terraform_workspace:
+        required: true
+        type: string
+      docker_image:
+        required: true
+        type: string
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  infrastructure_data:
+    runs-on: ubuntu-latest
+    timeout-minutes: 80
+    environment: ${{ inputs.environment }}
+    env:
+      TF_WORKSPACE: ${{ inputs.terraform_workspace }}
+    permissions:
+      contents: read
+      deployments: write
+    outputs:
+      services_to_deploy: ${{ steps.terraform.outputs.queue_workers_services }}
+      ecs_cluster: ${{ steps.terraform.outputs.ecs_cluster }}
+      aws_region: ${{ steps.terraform.outputs.aws_region }}
+    steps:
+      - run: echo "Deploying ${{ inputs.service_name }} to ${{ inputs.terraform_workspace }} And Docker Tag ${{ inputs.docker_image }}"
+      - name: Checkout cloud infra
+        uses: actions/checkout@master
+        with:
+          repository: novuhq/cloud-infra
+          token: ${{ secrets.GH_PACKAGES }}
+          path: cloud-infra
+
+      - name: Terraform setup
+        uses: hashicorp/setup-terraform@v1
+        with:
+          cli_config_credentials_token: ${{ secrets.TF_API_TOKEN }}
+          terraform_version: 1.5.5
+          terraform_wrapper: false
+
+      - name: Terraform Init
+        working-directory: cloud-infra/terraform/novu/aws
+        run: terraform init
+
+      - name: Terraform get output
+        working-directory: cloud-infra/terraform/novu/aws
+        id: terraform
+        run: |
+          echo "queue_workers_services=$(terraform output -json queue_workers_services)" >> $GITHUB_OUTPUT
+          echo "ecs_cluster=$(terraform output -json worker_ecs_cluster | jq -r .)" >> $GITHUB_OUTPUT
+          echo "aws_region=$(terraform output -json aws_region | jq -r .)" >> $GITHUB_OUTPUT
+
+
+  deploy_worker_queue:
+    needs: infrastructure_data
+    runs-on: ubuntu-latest
+    timeout-minutes: 80
+    environment: ${{ inputs.environment }}
+    env:
+      TF_WORKSPACE: ${{ inputs.terraform_workspace }}
+    permissions:
+      contents: read
+      deployments: write
+    strategy:
+      matrix:
+        worker: ${{fromJson(needs.infrastructure_data.outputs.services_to_deploy)}}
```

Comment:
This json is located in terraform and dynamically extracts all workers we have there